### ISSUE TITLE
[Bug]: Desync errors since 14.0

### ISSUE THREAD

**Issue by efess** [OpenTTD/OpenTTD#12502]
### Version of OpenTTD

14.0 Linux Generic AMD64 running in Ubuntu 22 Container

### Expected result

No Desyncs

### Actual result

Sporadic desyncs from players. Usually affects multiple players at once, not all however.

### Steps to reproduce

This is happening on Reddit Server 1
Config is https://gitlab.com/reddit-openttd/server1/-/tree/main/src/config
No GRF's loaded

**Comment by TrueBrain** [comment]
Would you have a savegame from either the server or client from just before or just after a desync?

And when abouts do desyncs happen .. after 10 minutes? an hour? 10 hours? Just always randomly?

**Comment by efess** [comment]
I'll work on getting some debug data today. Here's a dump of the log's to get an idea for timing (grep'd for desync)

```
[2024-04-13 23:02:40] dbg: [net:1] 'Terkala' reported an error and is closing its connection: desync error
[2024-04-13 23:02:40] dbg: [net:1] 'LIBYA' reported an error and is closing its connection: desync error
[2024-04-13 23:02:43] dbg: [net:1] 'Sektrex' reported an error and is closing its connection: desync error
[2024-04-14 07:01:06] dbg: [net:1] 'Tib' reported an error and is closing its connection: desync error
[2024-04-14 07:01:06] dbg: [net:1] 'Turner' reported an error and is closing its connection: desync error
[2024-04-14 07:01:06] dbg: [net:1] 'Gahare' reported an error and is closing its connection: desync error
[2024-04-14 07:01:06] dbg: [net:1] 'Apple_iHax' reported an error and is closing its connection: desync error
[2024-04-14 07:01:06] dbg: [net:1] 'player223' reported an error and is closing its connection: desync error
[2024-04-14 07:01:06] dbg: [net:1] 'ravenzar' reported an error and is closing its connection: desync error
[2024-04-14 07:01:06] dbg: [net:1] 'kaka' reported an error and is closing its connection: desync error
[2024-04-14 07:01:06] dbg: [net:1] 'Rutubet' reported an error and is closing its connection: desync error
[2024-04-14 13:32:36] dbg: [net:1] 'Lama' reported an error and is closing its connection: desync error
[2024-04-14 13:32:36] dbg: [net:1] 'Caralho' reported an error and is closing its connection: desync error
[2024-04-14 13:32:36] dbg: [net:1] 'Apple_iHax' reported an error and is closing its connection: desync error
[2024-04-14 13:32:36] dbg: [net:1] 'StarBurtS' reported an error and is closing its connection: desync error
[2024-04-14 13:32:36] dbg: [net:1] 'damyen' reported an error and is closing its connection: desync error
[2024-04-14 13:32:36] dbg: [net:1] 'Sugary' reported an error and is closing its connection: desync error
[2024-04-14 20:44:48] dbg: [net:1] 'Chuggers' reported an error and is closing its connection: desync error
[2024-04-14 20:44:48] dbg: [net:1] 'Aersling' reported an error and is closing its connection: desync error
[2024-04-14 20:44:48] dbg: [net:1] 'cizonit' reported an error and is closing its connection: desync error
[2024-04-14 20:44:48] dbg: [net:1] 'ravenzar' reported an error and is closing its connection: desync error
[2024-04-14 20:44:48] dbg: [net:1] 'Kangoo' reported an error and is closing its connection: desync error
[2024-04-14 20:44:48] dbg: [net:1] 'Nasu CFR' reported an error and is closing its connection: desync error
[2024-04-14 21:29:16] dbg: [net:1] 'Aersling' reported an error and is closing its connection: desync error
[2024-04-14 21:29:16] dbg: [net:1] 'Caralho' reported an error and is closing its connection: desync error
[2024-04-14 21:29:16] dbg: [net:1] 'Cizonit' reported an error and is closing its connection: desync error
[2024-04-14 21:29:16] dbg: [net:1] 'DEVU' reported an error and is closing its connection: desync error
[2024-04-14 21:36:00] dbg: [net:1] 'Aersling' reported an error and is closing its connection: desync error
[2024-04-14 21:36:00] dbg: [net:1] 'Chuggers' reported an error and is closing its connection: desync error
[2024-04-14 21:36:00] dbg: [net:1] 'Cizonit' reported an error and is closing its connection: desync error
[2024-04-14 21:36:00] dbg: [net:1] 'Lama' reported an error and is closing its connection: desync error
[2024-04-14 21:36:00] dbg: [net:1] 'DEVU' reported an error and is closing its connection: desync error
[2024-04-14 21:36:00] dbg: [net:1] 'TadasLT' reported an error and is closing its connection: desync error
[2024-04-14 21:36:00] dbg: [net:1] 'ravenzar' reported an error and is closing its connection: desync error
[2024-04-14 21:36:00] dbg: [net:1] 'Kangoo' reported an error and is closing its connection: desync error
[2024-04-14 23:26:29] dbg: [net:1] 'ministorm' reported an error and is closing its connection: desync error
[2024-04-14 23:26:29] dbg: [net:1] 'Chuggers' reported an error and is closing its connection: desync error
[2024-04-14 23:26:29] dbg: [net:1] 'Bursa' reported an error and is closing its connection: desync error
[2024-04-14 23:28:39] dbg: [net:1] 'Chuggers' reported an error and is closing its connection: desync error
[2024-04-14 23:28:39] dbg: [net:1] 'ministorm' reported an error and is closing its connection: desync error
[2024-04-14 23:28:39] dbg: [net:1] 'Daseem Ultor' reported an error and is closing its connection: desync error
[2024-04-15 00:00:23] dbg: [net:1] 'ministorm' reported an error and is closing its connection: desync error
[2024-04-15 00:00:23] dbg: [net:1] 'davinki' reported an error and is closing its connection: desync error
[2024-04-15 00:00:23] dbg: [net:1] 'Light Air' reported an error and is closing its connection: desync error
[2024-04-15 00:00:23] dbg: [net:1] 'Tib' reported an error and is closing its connection: desync error
[2024-04-15 00:00:23] dbg: [net:1] 'Aersling' reported an error and is closing its connection: desync error
[2024-04-15 00:00:23] dbg: [net:1] 'Lama' reported an error and is closing its connection: desync error
[2024-04-15 00:00:23] dbg: [net:1] 'DEVU' reported an error and is closing its connection: desync error
[2024-04-15 00:00:23] dbg: [net:1] 'LIBYA' reported an error and is closing its connection: desync error
[2024-04-15 00:11:53] dbg: [net:1] 'Vansta' reported an error and is closing its connection: desync error
[2024-04-15 00:11:53] dbg: [net:1] 'ravenzar' reported an error and is closing its connection: desync error
[2024-04-15 00:11:56] dbg: [net:1] 'Chuggers' reported an error and is closing its connection: desync error
[2024-04-15 03:33:58] dbg: [net:1] 'Aersling' reported an error and is closing its connection: desync error
[2024-04-15 03:38:12] dbg: [net:1] 'Chuggers' reported an error and is closing its connection: desync error
[2024-04-15 03:38:12] dbg: [net:1] 'CyberPotato' reported an error and is closing its connection: desync error
[2024-04-15 03:38:12] dbg: [net:1] 'Light Air' reported an error and is closing its connection: desync error
[2024-04-15 03:38:12] dbg: [net:1] 'Tib' reported an error and is closing its connection: desync error
[2024-04-15 03:38:12] dbg: [net:1] 'Luser' reported an error and is closing its connection: desync error
[2024-04-15 03:41:19] dbg: [net:1] 'CyberPotato' reported an error and is closing its connection: desync error
[2024-04-15 03:41:19] dbg: [net:1] 'Aersling' reported an error and is closing its connection: desync error
[2024-04-15 03:41:19] dbg: [net:1] 'Light Air' reported an error and is closing its connection: desync error
[2024-04-15 03:41:22] dbg: [net:1] 'ravenzar #1' reported an error and is closing its connection: desync error
[2024-04-15 03:46:25] dbg: [net:1] 'Dogma' reported an error and is closing its connection: desync error
[2024-04-15 03:46:25] dbg: [net:1] 'Stew' reported an error and is closing its connection: desync error
[2024-04-15 03:51:50] dbg: [net:1] 'Devenu' reported an error and is closing its connection: desync error
[2024-04-15 22:07:24] dbg: [net:1] 'ministorm' reported an error and is closing its connection: desync error
[2024-04-15 22:07:24] dbg: [net:1] 'nickel' reported an error and is closing its connection: desync error
[2024-04-15 22:07:24] dbg: [net:1] 'Sektrex' reported an error and is closing its connection: desync error
[2024-04-15 22:07:24] dbg: [net:1] 'Icron' reported an error and is closing its connection: desync error
[2024-04-15 22:07:24] dbg: [net:1] 'yamaguchi' reported an error and is closing its connection: desync error
[2024-04-15 22:13:55] dbg: [net:1] 'chasekeith' reported an error and is closing its connection: desync error
[2024-04-15 22:13:55] dbg: [net:1] 'Icron' reported an error and is closing its connection: desync error
[2024-04-15 22:13:55] dbg: [net:1] 'Chuggers' reported an error and is closing its connection: desync error
[2024-04-15 22:13:55] dbg: [net:1] 'superman' reported an error and is closing its connection: desync error
[2024-04-15 22:13:55] dbg: [net:1] 'BurntPineapple' reported an error and is closing its connection: desync error
[2024-04-15 22:13:55] dbg: [net:1] 'Stack' reported an error and is closing its connection: desync error
[2024-04-15 22:13:55] dbg: [net:1] 'DJDAN' reported an error and is closing its connection: desync error
[2024-04-15 22:13:55] dbg: [net:1] 'yamaguchi' reported an error and is closing its connection: desync error
[2024-04-15 22:41:16] dbg: [net:1] 'Sektrex' reported an error and is closing its connection: desync error
[2024-04-15 22:41:16] dbg: [net:1] 'Johnnie Walker #1' reported an error and is closing its connection: desync error
[2024-04-16 01:46:44] dbg: [net:1] 'AndyMissed' reported an error and is closing its connection: desync error
[2024-04-16 01:46:44] dbg: [net:1] 'T0mKat' reported an error and is closing its connection: desync error
[2024-04-16 01:46:44] dbg: [net:1] 'Helmholtz' reported an error and is closing its connection: desync error
[2024-04-16 01:46:44] dbg: [net:1] 'Sektrex' reported an error and is closing its connection: desync error
[2024-04-16 01:46:44] dbg: [net:1] 'Dogma AFK' reported an error and is closing its connection: desync error
[2024-04-16 01:46:44] dbg: [net:1] 'Caralho_AFK' reported an error and is closing its connection: desync error
[2024-04-16 01:46:44] dbg: [net:1] 'ray' reported an error and is closing its connection: desync error
[2024-04-16 01:46:44] dbg: [net:1] 'Johnnie Walker' reported an error and is closing its connection: desync error
[2024-04-16 01:49:44] dbg: [net:1] 'AndyMissed' reported an error and is closing its connection: desync error
[2024-04-16 01:49:44] dbg: [net:1] 'Sektrex' reported an error and is closing its connection: desync error
[2024-04-16 01:49:44] dbg: [net:1] 'Helmholtz' reported an error and is closing its connection: desync error
[2024-04-16 01:49:44] dbg: [net:1] 'ray' reported an error and is closing its connection: desync error
```

**Comment by efess** [comment]
Here's an archive of autosaves around the time of the logs above. These aren't very granular - they take place every game year.

https://drive.google.com/file/d/1oDIOQARxFymHg8joq0zA0PtSkkPHuGu9/view?usp=sharing

**Comment by efess** [comment]
Got one.  Logs during this time:

```
[2024-04-16 19:38:03] dbg: [net:3] [server] Client connected from ***.***.***.*** on frame 718998
*** sockmeistr has joined the game (Client #97)
[2024-04-16 19:38:07] dbg: [net:3] [server] Client #97 (***.***.***.***) joined as sockmeistr
[Private] To sockmeistr: Welcome to the reddit1 vanilla server
[Private] To sockmeistr: Hosted by the OpenTTD Community
[Private] To sockmeistr: Please be respectful to others
[2024-04-16 19:38:07] dbg: [net:3] [admin] Rcon command from 'restmin' (0): clients
[Private] To sockmeistr: Type !help for help and !rules for the house rules
[Private] To sockmeistr: Remember to set your company password!
[2024-04-16 19:38:07] dbg: [net:3] [admin] Rcon command from 'restmin' (0): clients
[2024-04-16 19:38:08] dbg: [net:3] [admin] Rcon command from 'restmin' (0): clients
*** DJDAN has left the game (leaving)
[2024-04-16 19:38:08] dbg: [net:3] [server] Client #93 closed connection
[2024-04-16 19:38:09] dbg: [net:3] [admin] Rcon command from 'restmin' (0): companies
[2024-04-16 19:38:09] dbg: [net:3] [admin] Rcon command from 'restmin' (0): clients
[2024-04-16 19:38:41] dbg: [net:3] [admin] Rcon command from 'restmin' (0): companies
[2024-04-16 19:38:41] dbg: [net:3] [admin] Rcon command from 'restmin' (0): clients
[2024-04-16 19:39:11] dbg: [net:3] [admin] Rcon command from 'restmin' (0): companies
[2024-04-16 19:39:11] dbg: [net:3] [admin] Rcon command from 'restmin' (0): clients
[2024-04-16 19:39:43] dbg: [net:3] [admin] Rcon command from 'restmin' (0): companies
[2024-04-16 19:39:43] dbg: [net:3] [admin] Rcon command from 'restmin' (0): clients
[2024-04-16 19:39:49] dbg: [net:1] 'Elm' reported an error and is closing its connection: desync error
[2024-04-16 19:39:49] dbg: [net:3] [server] Client #72 closed connection
*** Elm has left the game (desync error)
[2024-04-16 19:39:49] dbg: [net:1] 'ChewChoo' reported an error and is closing its connection: desync error
[2024-04-16 19:39:49] dbg: [net:3] [server] Client #26 closed connection
[2024-04-16 19:39:49] dbg: [net:1] 'ray' reported an error and is closing its connection: desync error
[2024-04-16 19:39:49] dbg: [net:3] [server] Client #85 closed connection
*** ChewChoo has left the game (desync error)
*** ray has left the game (desync error)
[2024-04-16 19:39:49] dbg: [net:1] 'Lama' reported an error and is closing its connection: desync error
[2024-04-16 19:39:49] dbg: [net:3] [server] Client #18 closed connection
*** Lama has left the game (desync error)
[2024-04-16 19:39:49] dbg: [net:1] 'John Download' reported an error and is closing its connection: desync error
*** John Download has left the game (desync error)
[2024-04-16 19:39:49] dbg: [net:3] [server] Client #83 closed connection
[2024-04-16 19:39:49] dbg: [net:1] 'OxO' reported an error and is closing its connection: desync error
*** OxO has left the game (desync error)
[2024-04-16 19:39:49] dbg: [net:3] [server] Client #92 closed connection
[2024-04-16 19:39:54] dbg: [net:3] [server] Client connected from ***.***.***.*** on frame 722465
*** ray has joined the game (Client #98)
[2024-04-16 19:40:00] dbg: [net:3] [server] Client #98 (***.***.***.***) joined as ray
[Private] To ray: Welcome to the reddit1 vanilla server
[2024-04-16 19:40:00] dbg: [net:3] [admin] Rcon command from 'restmin' (0): clients
[Private] To ray: Hosted by the OpenTTD Community
[Private] To ray: Please be respectful to others
[Private] To ray: Type !help for help and !rules for the house rules
[Private] To ray: Remember to set your company password!
[2024-04-16 19:40:00] dbg: [net:3] [admin] Rcon command from 'restmin' (0): clients
[2024-04-16 19:40:00] dbg: [net:3] [admin] Rcon command from 'restmin' (0): clients
*** ray has joined company #1
[2024-04-16 19:40:09] dbg: [net:3] [server] Client #97 closed connection
*** sockmeistr has left the game (leaving)
```

```
-rw-r--r-- 1 openttd openttd  2805332 Apr 16 19:38 dmp_cmds_d601a577_000af5e0.sav
-rw-r--r-- 1 openttd openttd  2777272 Apr 16 19:40 dmp_cmds_d601a577_000af600.sav
```

[desync_artifacts.zip](https://github.com/OpenTTD/OpenTTD/files/15001936/desync_artifacts.zip)

**Comment by efess** [comment]
Few more requested `.sav`s:

```-rw-r--r-- 1 openttd openttd  1428908 Apr 16 14:47 dmp_cmds_d601a577_000ad560.sav
-rw-r--r-- 1 openttd openttd  1383136 Apr 16 14:48 dmp_cmds_d601a577_000ad580.sav
-rw-r--r-- 1 openttd openttd  1436392 Apr 16 14:49 dmp_cmds_d601a577_000ad5a0.sav

-rw-r--r-- 1 openttd openttd  1440948 Apr 16 15:06 dmp_cmds_d601a577_000ad780.sav
-rw-r--r-- 1 openttd openttd  1489952 Apr 16 15:07 dmp_cmds_d601a577_000ad7a0.sav
-rw-r--r-- 1 openttd openttd  1504168 Apr 16 15:08 dmp_cmds_d601a577_000ad7c0.sav
```

[evidence_14-48_15-07.zip](https://github.com/OpenTTD/OpenTTD/files/15002196/evidence_14-48_15-07.zip)

**Comment by efess** [comment]
Dumps around 19:23

```
-rw-r--r-- 1 openttd openttd  2733552 Apr 16 19:22 dmp_cmds_d601a577_000af440.sav
-rw-r--r-- 1 openttd openttd  2736668 Apr 16 19:23 dmp_cmds_d601a577_000af460.sav
-rw-r--r-- 1 openttd openttd  2709044 Apr 16 19:24 dmp_cmds_d601a577_000af480.sav
```
[evidence_19-23.zip](https://github.com/OpenTTD/OpenTTD/files/15002259/evidence_19-23.zip)

**Comment by TrueBrain** [comment]
Been doing some replays (with the excellent desync docs):

replaying from start: no desync
replaying from ad7c0: no desync
replaying from af400: desync at 000af5fe, with cmdf replay at 000af5f9
replaying from af800: same as af400
replaying from af5e0: no desync

**Comment by efess** [comment]
Save files from around the desync plus latest `commands-out.log`

```
-rw-r--r-- 1 openttd openttd  3615612 Apr 17 01:02 dmp_cmds_d601a577_000b12c0.sav
-rw-r--r-- 1 openttd openttd  3615820 Apr 17 01:04 dmp_cmds_d601a577_000b12e0.sav
-rw-r--r-- 1 openttd openttd  3599428 Apr 17 01:05 dmp_cmds_d601a577_000b1300.sav
-rw-r--r-- 1 openttd openttd  3615816 Apr 17 01:07 dmp_cmds_d601a577_000b1320.sav
```

Log entries...

```
*** efess has left the game (desync error)
[2024-04-17 01:04:41] dbg: [net:1] 'efess' reported an error and is closing its connection: desync error
[2024-04-17 01:04:41] dbg: [net:3] [server] Client #3 closed connection
*** chasekeith has left the game (desync error)
[2024-04-17 01:04:41] dbg: [net:1] 'chasekeith' reported an error and is closing its connection: desync error
[2024-04-17 01:04:41] dbg: [net:3] [server] Client #152 closed connection
[2024-04-17 01:04:41] dbg: [net:1] 'ChewChoo' reported an error and is closing its connection: desync error
[2024-04-17 01:04:41] dbg: [net:3] [server] Client #105 closed connection
[2024-04-17 01:04:41] dbg: [net:1] 'superman' reported an error and is closing its connection: desync error
[2024-04-17 01:04:41] dbg: [net:3] [server] Client #37 closed connection
[2024-04-17 01:04:41] dbg: [net:1] 'nickel' reported an error and is closing its connection: desync error
[2024-04-17 01:04:41] dbg: [net:3] [server] Client #121 closed connection
[2024-04-17 01:04:41] dbg: [net:1] 'Jinty Steam' reported an error and is closing its connection: desync error
[2024-04-17 01:04:41] dbg: [net:3] [server] Client #168 closed connection
*** ChewChoo has left the game (desync error)
*** superman has left the game (desync error)
*** nickel has left the game (desync error)
*** Jinty Steam has left the game (desync error)
[2024-04-17 01:04:41] dbg: [net:1] 'ray' reported an error and is closing its connection: desync error
[2024-04-17 01:04:41] dbg: [net:3] [server] Client #147 closed connection
*** ray has left the game (desync error)
[2024-04-17 01:04:41] dbg: [net:1] 'pinworm' reported an error and is closing its connection: desync error
*** pinworm has left the game (desync error)

```
[second_desync_cmds_log.zip](https://github.com/OpenTTD/OpenTTD/files/15004416/second_desync_cmds_log.zip)

**Comment by Loafdude** [comment]
I also am getting these sync errors playing on Reddit Server #1. 
Please let me know if and how I can assist.

**Comment by PeterN** [comment]
Resolved by #12512
